pkg/solver/controller.go

```diff
@@ -321,6 +322,16 @@ func (controller *SolverController) addResourceOffer(resourceOffer data.Resource
 	}
 	resourceOffer.ID = id
 
+	// Check the resource provider's ETH balance
+	balance, err := controller.web3SDK.GetBalance(resourceOffer.ResourceProvider)
+	if err != nil {
+		return nil, err
+	}
+	// If the balance is 0, don't add the resource offer
+	if balance.Cmp(big.NewInt(0)) == 0 {
```

Comment:
This is a good first pass, but shouldn't we ultimately be checking that the RP not-only has non-zero balance, but has enough balance for the transaction?

Reply:
We should be we would need to run an estimation based on the job offers available which can vary. 

What we can do is add the `resourceOffer.DefaultPricing` values and use that as a comparison here.

Reply:
In order to estimate the transaction, we'll need the actual transaction object that will be sent to the smart contract to estimate the gas cost. This may require refactoring the `resourceOffer` struct or something similar. I'll take a look at `resourceOffer.InstructionPrice` to see if it can be used for balance checking.

Reply:
I think we can use this: `resourceOffer.DefaultPricing.InstructionPrice`